### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by cbielow**
## Description

mostly in RangeManager, which repeatedly shows up due to being a templated class

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/include/OpenMS/KERNEL/RangeManager.h [hunk 1]
```diff
@@ -113,13 +113,13 @@ namespace OpenMS
     /// is @p value within [min, max]?
     bool contains(const double value) const
     {
-      return (min_ <= value) & (value <= max_);
+      return uint8_t(min_ <= value) & uint8_t(value <= max_); // using && leads to branches on all compilers in Debug and in Release on MVSC
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/KERNEL/RangeManager.h:116]: Note:
```
if ( (min_ <= value) && (value <= max_) ) return true; else false;
```
should AFAIK trigger the short circuit

Comment (cbielow) [reply src/openms/include/OpenMS/KERNEL/RangeManager.h:116]: nope. With that solution: Clang at O1 still has branches, GCC at O0 also has branches. MSVC always branches.

Comment (timosachsenberg) [reply src/openms/include/OpenMS/KERNEL/RangeManager.h:116]: hmm strange... maybe it knows that branch prediction will always make the right choice anyway so it doesn't have performance impact?

Surprising... I thought in if (not a normal boolean expression) the second term is not allowed to be evaluated if the first one is false.

**Comment by jpfeuffer** [comment]
Can you check the delete with non-virtual destructor warnings of LayerBase? Are they fixed too?

**Comment by cbielow** [comment]
> Can you check the delete with non-virtual destructor warnings of LayerBase? Are they fixed too?

there is no such class.
But I fixed a few more warnings.
Should be good to merge.

**Comment by cbielow** [comment]
will merge to see what is left in the next nightly build.
Test failures on Windows are unrelated (XTandem and File_test)

**Pull request merged**